src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,130 @@
+    void setUp() {
+        validUser = new User(VALID_USER_LOGIN, VALID_USER_PASSWORD, VALID_USER_AGE);
+    }
+
+    @Test
+    void register_validUser_Ok() {
+        assertEquals(validUser,registrationService.register(validUser));
+        assertNotNull(validUser.getId());
+        assertTrue(Storage.people.contains(validUser));
+    }
+
+    @Test
+    void register_lessThanMinAge_NotOk() {
+        validUser.setAge(VALID_USER_AGE - 1);
+        assertThrows(RuntimeException.class, () -> {
+            registrationService.register(validUser);
+        });
+    }
+
+    @Test
+    void register_negativeAge_NotOk() {
+        validUser.setAge(-18);
+        assertThrows(RuntimeException.class, () -> {
+            registrationService.register(validUser);
+        });
+    }
+
+    @Test
+    void register_ageMoreThanMin_Ok() {
+        validUser.setAge(46);
+        assertEquals(validUser,registrationService.register(validUser));
+        assertNotNull(validUser.getId());
+        assertTrue(Storage.people.contains(validUser));
+    }
+
+    @Test
+    void register_lessThanMinPassword_NotOk() {
+        validUser.setPassword("pass");
+        assertThrows(RuntimeException.class, () -> {
+            registrationService.register(validUser);
+        });
+    }
+
+    @Test
+    void register_minLengthPassword_Ok() {
+        validUser.setPassword("123456");
+        assertEquals(validUser,registrationService.register(validUser));
+        assertNotNull(validUser.getId());
+        assertTrue(Storage.people.contains(validUser));
+    }
+
+    @Test
+    void register_moreThanMinPassword_Ok() {
+        validUser.setPassword("123456password654321");
+        assertEquals(validUser,registrationService.register(validUser));
+        assertNotNull(validUser.getId());
+        assertTrue(Storage.people.contains(validUser));
+    }
+
+    @Test
+    void register_UserAlreadyExist_NotOk() {
+        registrationService.register(validUser);
+        assertThrows(RuntimeException.class, () -> {
+            registrationService.register(validUser);
+        });
+    }
+
+    @Test
+    void register_nullUser_notOk() {
+        assertThrows(RuntimeException.class, () -> {
+            registrationService.register(null);
+        });
+    }
+
+    @Test
+    void register_nullAge_notOk() {
+        validUser.setAge(null);
+        assertThrows(RuntimeException.class, () -> {
+            registrationService.register(validUser);
+        });
+    }
+
+    @Test
+    void register_nullPassword_notOk() {
+        validUser.setPassword(null);
+        assertThrows(RuntimeException.class, () -> {
+            registrationService.register(validUser);
+        });
+    }
+
+    @Test
+    void register_nullLogin_notOk() {
+        validUser.setLogin(null);
+        assertThrows(RuntimeException.class, () -> {
+            registrationService.register(validUser);
+        });
+    }
+
+    @AfterEach
+    void tearDown() {
+        Storage.people.clear();
+    }
```

Comment:
```suggestion
    @AfterEach
    void tearDown() {
        Storage.people.clear();
    }
```
Perhaps you need to put this annotation after BeforeEach